scipy/signal/_fir_filter_design.py

```diff
@@ -426,26 +434,27 @@ def firwin(numtaps, cutoff, *, width=None, window='hamming', pass_zero=True,
                              f'{pass_zero}')
     pass_zero = bool(operator.index(pass_zero))  # ensure bool-like
 
-    pass_nyquist = bool(cutoff.size & 1) ^ pass_zero
+    pass_nyquist = bool(xp_size(cutoff) & 1) ^ pass_zero
     if pass_nyquist and numtaps % 2 == 0:
         raise ValueError("A filter with an even number of coefficients must "
                          "have zero response at the Nyquist frequency.")
 
     # Insert 0 and/or 1 at the ends of cutoff so that the length of cutoff
     # is even, and each pair in cutoff corresponds to passband.
-    cutoff = np.hstack(([0.0] * pass_zero, cutoff, [1.0] * pass_nyquist))
+    cutoff = xp.concat((xp.zeros(int(pass_zero)), cutoff, xp.ones(int(pass_nyquist))))
+
 
     # `bands` is a 2-D array; each row gives the left and right edges of
     # a passband.
-    bands = cutoff.reshape(-1, 2)
+    bands = xp.reshape(cutoff, (-1, 2))
 
     # Build up the coefficients.
     alpha = 0.5 * (numtaps - 1)
-    m = np.arange(0, numtaps) - alpha
+    m = xp.arange(0, numtaps) - alpha
     h = 0
     for left, right in bands:
-        h += right * sinc(right * m)
-        h -= left * sinc(left * m)
+        h += right * xp.sinc(right * m)
```

Comment:
```suggestion
        h += right * xpx.sinc(right * m, xp=xp)
```
Do we need to use xpx here?